src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -754,6 +758,8 @@ public static ServerInfo GetServerVersion(IDbConnection connection)
                 delegate (IDataReader reader)
                 {
                     reader.Read();
+                    serverInfo.Options = new Dictionary<string, object>();
```

Comment:
Do we need to construct serverInfo.Options even when reader.FieldCount <= 6?, could serverInfo.Options stay null in those scenarios? or does it need to be an empty Dictionary?